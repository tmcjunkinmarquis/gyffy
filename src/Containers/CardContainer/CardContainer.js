import React from 'react';
import { Switch, Route, withRouter } from 'react-router-dom';
import FuelStationList from '../FuelStationList/FuelStationList';
import ElectricStationList from '../ElectricStationList/ElectricStationsList';
import LpgStationList from '../LpgStationList/LpgStationList';
import DetailsCard from '../DetailsCard/DetailsCard';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

export const CardContainer = (props)=>{
  return (
    <Switch>
      <Route
        exact path="/all"
        component={FuelStationList}
      />
      <Route
        exact path="/electric"
        component={ElectricStationList}
      />
      <Route
        exact path="/lpg"
        component={LpgStationList}
      />
      <Route
        path="/details/:id"
        render={({match})=>{
          const foundStation = props.stations.find((station)=>{
            return station.id === Number(match.params.id);
          });
          return (
            <DetailsCard foundStation={foundStation} />
          );
        }}
      />
    </Switch> 
  );
};

export const mapStateToProps = (state)=>({
  stations: state.stations
});

export default withRouter(connect(mapStateToProps)(CardContainer));

CardContainer.propTypes = {
  stations: PropTypes.array
};