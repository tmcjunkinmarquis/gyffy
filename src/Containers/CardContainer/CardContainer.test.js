import React from 'react';
import { shallow } from  'enzyme';
import { CardContainer, mapStateToProps } from './CardContainer';


describe('CardContainer Container', () => {
  
  describe('CardContainer Component', () => {
      let wrapper;
    beforeEach(()=>{
      wrapper = shallow(<CardContainer />)
    })

    it('should match a snapshot', () => {
      expect(wrapper).toMatchSnapshot()
    });
  });

  describe('mapStateToProps', () => {
    
    it('should return an object with the station', () => {
      // Setup
      const mockState = {
        stations: {name: 'bubbas'}
      };
      const expected = {
        stations: { name: 'bubbas' }
      };

      // Execution
      const mappedProps = mapStateToProps(mockState);

      // Expectation
      expect(mappedProps).toEqual(expected);
    });
  });

});



