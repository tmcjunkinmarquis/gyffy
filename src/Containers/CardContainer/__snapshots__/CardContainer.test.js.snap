// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer Container CardContainer Component should match a snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Route
          component={[Function]}
          exact={true}
          path="/all"
/>,
        <Route
          component={[Function]}
          exact={true}
          path="/electric"
/>,
        <Route
          component={[Function]}
          exact={true}
          path="/lpg"
/>,
        <Route
          path="/details/:id"
          render={[Function]}
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/all",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/electric",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/lpg",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "path": "/details/:id",
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Route
            component={[Function]}
            exact={true}
            path="/all"
/>,
          <Route
            component={[Function]}
            exact={true}
            path="/electric"
/>,
          <Route
            component={[Function]}
            exact={true}
            path="/lpg"
/>,
          <Route
            path="/details/:id"
            render={[Function]}
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/all",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/electric",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/lpg",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "path": "/details/:id",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
